images/hook/entrypoint.sh

```diff
@@ -20,8 +20,32 @@ if [ $1 = "update" ]; then
     echo "---> Deleting old deployment 'kapacitor'"
     rig delete deployments/kapacitor --resource-namespace=monitoring --force
 
+    echo "---> Removing last applied configuration from resources"
+    for deployment in grafana kube-state-metrics prometheus-adapter prometheus-operator watcher
+    do
+      # || true is needed in case resources does not have last applied configuration
+      kubectl --namespace=monitoring patch deployments.apps $deployment --type=json -p='[{"op": "replace", "path": "/metadata/managedFields", "value": [{}]}]' || true
+      kubectl --namespace=monitoring patch deployments.apps $deployment --type=json -p='[{"op": "remove", "path": "/metadata/annotations/kubectl.kubernetes.io~1last-applied-configuration"}]' || true
```

Comment:
Can you explain the syntax of `~1` (if that's not a typo):
```
/metadata/annotations/kubectl.kubernetes.io~1last-applied-configuration
```

Reply:
`~` is the way to replace `/` in annotation key name:
https://stackoverflow.com/a/55575415
https://tools.ietf.org/html/rfc6901#section-3

Reply:
wow, that's some great design on the json pointer - make it even more unreadable. Ok, I see - so basically the annotation key is encoded with a different encoder that implements RFC6901 and turns `/` into `~1`.